main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -3178,22 +3251,32 @@ object Weeder2 {
     * IE. Calling on a tree of kind [[TreeKind.Ident]] is fine, but if the kind is not known avoid using [[tokenToIdent]].
     */
   private def tokenToIdent(tree: Tree)(implicit sctx: SharedContext): Name.Ident = {
-    tree.children.headOption match {
-      case Some(token@Token(_, _, _, _, sp1, sp2)) =>
-        Name.Ident(token.text, SourceLocation(isReal = true, sp1, sp2))
-      // If child is an ErrorTree, that means the parse already reported and error.
-      // We can avoid double reporting by returning a success here.
-      // Doing it this way is most resilient, but phases down the line might have trouble with this sort of thing.
-      case Some(t: Tree) if t.kind.isInstanceOf[TreeKind.ErrorTree] =>
-        val name = text(tree).mkString("")
-        Name.Ident(name, tree.loc)
-      case Some(t: Tree) if t.kind == TreeKind.CommentList =>
-        // We hit a misplaced comment.
+    tree.kind match {
```

Comment:
This is also an important change. Before this, it just checked directly the children. However, now it checks if the tree is an error node itself and handles it same way it would handle a child error node.

Reply:
This means the same disclaimer applies:
```
        // If child is an ErrorTree, that means the parse already reported and error.
        // We can avoid double reporting by returning a success here.
        // Doing it this way is most resilient, but phases down the line might have trouble with this sort of thing.
```